packages/esm-patient-medications-app/src/order-basket/order-basket-search/order-basket-search-results.component.tsx

```diff
@@ -0,0 +1,160 @@
+          caption={error?.message}
+        />
+      )}
+    </>
+  ) : null;
+}
+
+interface DrugSearchResultItemProps {
+  drug: Drug;
+  onSearchResultClicked: (searchResult: OrderBasketItem, directlyAddToBasket: boolean) => void;
+}
+
+const DrugSearchResultItem: React.FC<DrugSearchResultItemProps> = ({ drug, onSearchResultClicked }) => {
+  const isTablet = useLayoutType() === 'tablet';
+  const {
+    templates,
+    isLoading: isLoadingTemplates,
+    error: fetchingDrugOrderTemplatesError,
+  } = useDrugTemplate(drug?.uuid);
+  const { t } = useTranslation();
+  const config = useConfig() as ConfigObject;
+  const orderItems: Array<OrderBasketItem> = useMemo(
+    () =>
+      templates?.length
+        ? templates.map((template) => getTemplateOrderBasketItem(drug, config?.daysDurationUnit, template))
+        : [getTemplateOrderBasketItem(drug, config?.daysDurationUnit)],
+    [templates, drug, config?.daysDurationUnit],
+  );
+
+  const handleSearchResultClicked = (searchResult: OrderBasketItem, directlyAddToBasket: boolean) => {
+    onSearchResultClicked(searchResult, directlyAddToBasket);
+  };
+
+  return (
+    <>
+      {orderItems.map((orderItem, indx) => (
+        <ClickableTile
+          key={templates?.length ? templates[indx]?.uuid : drug?.uuid}
+          role="listitem"
+          className={isTablet ? `${styles.tabletSearchResultTile}` : `${styles.desktopSearchResultTile}`}
+          onClick={() => handleSearchResultClicked(orderItem, false)}
+        >
+          <div className={styles.searchResultTile}>
+            <div className={styles.searchResultTileContent}>
+              <p>
+                <strong>{drug?.display}</strong> {drug?.strength && <>&mdash; {drug?.strength}</>}{' '}
+                {drug?.dosageForm?.display && <>&mdash; {drug?.dosageForm?.display}</>}
+              </p>
+              {fetchingDrugOrderTemplatesError ? (
+                <p>
+                  <span className={styles.errorLabel}>
+                    {t('errorFetchingDrugOrderTemplates', 'Error fetching drug order templates')}
+                  </span>
+                </p>
+              ) : (
+                <p>
+                  {orderItem?.frequency?.value && <span className={styles.label01}>{orderItem?.frequency?.value}</span>}
+                  {orderItem?.route?.value && <span className={styles.label01}>&mdash; {orderItem?.route?.value}</span>}
+                </p>
+              )}
+            </div>
+            <Button
+              className={styles.addToBasketButton}
+              kind="ghost"
+              hasIconOnly={true}
+              renderIcon={(props) => <ShoppingCart size={16} {...props} />}
+              iconDescription={t('directlyAddToBasket', 'Immediately add to basket')}
+              onClick={() => handleSearchResultClicked(orderItem, true)}
+            />
+          </div>
+        </ClickableTile>
+      ))}
+    </>
+  );
+};
+
+const Skeleton = () => {
```

Comment:
At first, I was confused when I saw `Skeleton` being used and I hadn't seen where it was declared, I don't know if this doesn't bring an error or warning for using before initializing but I think it would be better to either have this at the top of the file for better readability or have it in another file and just import it.

Reply:
I updated the Skeleton's name